pytest_fixtures/component/subscription.py

```diff
@@ -24,39 +12,21 @@ def default_subscription(module_target_sat, module_org_with_manifest):
     return subscription[0]
 
 
-def subscribe_satellite(sat):
-    from robottelo.config import settings
-
-    if sat.os_version.major < 8:
-        release_version = f'{sat.os_version.major}Server'
-    else:
-        release_version = f'{sat.os_version.major}'
-    sat.register_contenthost(
-        org=None,
-        lce=None,
-        username=settings.subscription.rhn_username,
-        password=settings.subscription.rhn_password,
-        releasever=release_version,
-    )
-    result = sat.subscription_manager_attach_pool([settings.subscription.rhn_poolid])[0]
-    if 'Successfully attached a subscription' in result.stdout:
-        # extras is not in RHEL8: https://access.redhat.com/solutions/5331391
-        if sat.os_version.major < 8:
-            sat.enable_repo(f'rhel-{sat.os_version.major}-server-extras-rpms', force=True)
-        yield
-    else:
-        pytest.fail('Failed to attach system to pool. Aborting Test!.')
-    sat.unregister()
-    sat.remove_katello_ca()
-
-
 @pytest.fixture(scope='module')
-def module_subscribe_satellite(module_clean_rhsm, module_target_sat):
+def module_subscribe_satellite(module_target_sat):
     """subscribe satellite to cdn"""
-    subscribe_satellite(module_target_sat)
+    module_target_sat.register_to_cdn()
+    # Enable extras repo if os_version is RHEL7
+    if module_target_sat.os_version.major < 8:
+        module_target_sat.enable_repo(
+            f'rhel-{module_target_sat.os_version.major}-server-extras-rpms', force=True
+        )
 
```

Comment:
Yes, it would be teardown anyway, but just to be explicit, I would include a step to unregister, and as this fixture registers to CDN and not dogfood, `remove_katello_ca()` won't be required.

Reply:
Updated